src/platform/mt8186/platform.c

```diff
@@ -196,7 +196,9 @@ int platform_init(struct sof *sof)
 	/* initialize the host IPC mechanims */
 	ipc_init(sof);
 
-	/* TO-DO: dai init */
+	ret = dai_init(sof);
+	if (ret < 0)
+		return -ENODEV;
```

Comment:
why not just return ret?

Reply:
Thx, return ret is more reasonable, I'll change this